### PR TITLE
added rwa application support in auth0

### PR DESCRIPTION
**Pull request by rinzler99**

### CODE REVIEW

File: examples/auth0-react-example/app/.env.development [hunk 1]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_CLIENT_ID="BKPxkCtfC9gZ5dj-eg-W6yb5Xfr3XkxHuGZl2o2Bn8gKQ7UYike9Dh6c-_LaXlUN77x0cBoPwcSx-IVm0llVsLA"
+REACT_APP_VERIFIER="web3auth-firebase-demo"
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/.env.development:2]: change the name to : web3auth-auth0-demo

File: examples/auth0-react-example/app/README.md [hunk 2]
```diff
@@ -34,6 +34,9 @@ copy your clientId to env variable named `REACT_APP_CLIENT_ID` in env.developmen
 
 - Information about validating JWT token is already added while creating a custom verifier from [web3auth developer dashboard](https://dashboard.web3auth.io). So we just need to specify our verifier name while adding Openlogin Adapter to web3auth. Refer to `init()` function in `src/services/web3auth.tsx` file.
 
+## For RWA example you need to change following :
+1. Your domain name in loginRWA method
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/README.md:38]: load it from env variables.

File: examples/auth0-react-example/api/server.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const express = require('express')
+require("dotenv").config();
+const app = express();
+var request = require("request");
+
+port = process.env.PORT;
+
+//api to which your server listen for the requests coming from auth0
+app.get('/callback', (req, res) => {
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/api/server.js:9]: add eslint for your node js code.

File: examples/auth0-react-example/api/views/partials/header.ejs [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/api/views/partials/header.ejs:1]: are we using ejs files, if not remove them

File: examples/auth0-react-example/app/src/components/Main.tsx [hunk 5]
```diff
@@ -4,17 +4,17 @@ import { useWeb3Auth } from "../services/web3auth";
...
+  
+  const handleImplicitLogin = async () => {
     try {
-      setIsLoading(true);
-      await login(WALLET_ADAPTERS.OPENLOGIN,"jwt")
+      await login(WALLET_ADAPTERS.OPENLOGIN,"jwt");
     } finally {
-      setIsLoading(false);
     }
   }
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/src/components/Main.tsx:15]: remove this empty finally block.

File: examples/auth0-react-example/app/src/components/Main.tsx [hunk 6]
```diff
@@ -50,17 +50,24 @@ const Main = () => {
...
       <h3>Login With</h3>
-      <button onClick={()=> handleGoogleLogin()} className={styles.card}>
-        Email Passwordless
+      {appType === "SPA"?
+      <button onClick={()=> handleImplicitLogin()} className={styles.card}>
+        Implicit Flow(SPA)
+      </button>:
+      <button onClick={(e) => {
+      e.preventDefault();
+      window.location.href="https://torus-test.auth0.com/authorize?scope=openid&response_type=code&client_id=FX0BwYwDtD6p0yTOjjIykjLbtxXszfkR&redirect_uri=http://localhost:3001/callback&state=STATE&connection=github"
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/src/components/Main.tsx:65]: load this domain from env variables. and also load redirect_uri.

File: examples/auth0-react-example/app/src/services/web3authRWA.tsx [hunk 7]
```diff
@@ -14,7 +14,7 @@ export interface IWeb3AuthContext {
   isLoading: boolean;
   user: unknown;
   chain: string;
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/src/services/web3authRWA.tsx:16]: this file is not being used, remove it

File: examples/auth0-react-example/app/src/components/MainRWA.tsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import Loader from "./Loader";
+import styles from "../styles/Home.module.css";
+import {useLocation} from "react-router-dom";
+
+
+const Main = () => {
+  const { provider, loginRWA, logout, getUserInfo, getAccounts, getBalance, signMessage, isLoading, signTransaction, signAndSendTransaction, chain, web3Auth, setIsLoading } = useWeb3Auth();
+  const search = useLocation().search;
+  const jwt = new URLSearchParams(search).get('token');
+  const token = jwt == null ? "" : jwt.toString();
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/src/components/MainRWA.tsx:13]: toString() is not required

**Review by himanshuchawla009 — CHANGES_REQUESTED** [review]

File: examples/auth0-react-example/app/.env.development (hunk 1 above)
Comment (rinzler99) [reply examples/auth0-react-example/app/.env.development]: changed

File: examples/auth0-react-example/api/server.js (hunk 3 above)
Comment (rinzler99) [reply examples/auth0-react-example/api/server.js:9]: added eslint for nodejs code.

File: examples/auth0-react-example/api/views/partials/header.ejs (hunk 4 above)
Comment (rinzler99) [reply examples/auth0-react-example/api/views/partials/header.ejs]: they were removed locally , done same on remote repo

File: examples/auth0-react-example/app/src/components/Main.tsx (hunk 5 above)
Comment (rinzler99) [reply examples/auth0-react-example/app/src/components/Main.tsx:22]: removed

File: examples/auth0-react-example/app/src/components/Main.tsx (hunk 6 above)
Comment (rinzler99) [reply examples/auth0-react-example/app/src/components/Main.tsx]: done

File: examples/auth0-react-example/app/src/components/MainRWA.tsx (hunk 8 above)
Comment (rinzler99) [reply examples/auth0-react-example/app/src/components/MainRWA.tsx]: done

File: examples/auth0-react-example/api/.env.example [hunk 9]
```diff
@@ -0,0 +1,6 @@
+AUTH_URL = https://torus-test.auth0.com/oauth/token
+PORT = 3001
+CLIENT_ID = FX0BwYwDtD6p0yTOjjIykjLbtxXszfkR 
+CLIENT_SECRET =  TgwKmD3Jd9iXQu_Fhj0w1smrm32NSgHQ3gzKBY_K6YTU3klQemME7UGxpbr3DlC3
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/api/.env.example:4]: remove secret !

File: examples/auth0-react-example/app/src/components/Main.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+import styles from "../styles/Home.module.css";
+
+const Main = ({isJWT, appType}:{isJWT:boolean, appType: string}) => {
+  const { provider, login, logout, getUserInfo, getAccounts, getBalance, signMessage, isLoading, signTransaction, signAndSendTransaction, web3Auth, chain } = useWeb3Auth();
+  const rwaURL = process.env.REACT_APP_AUTH0_DOMAIN+
+                "/authorize?scope=openid&response_type=code&client_id="+
+                process.env.REACT_APP_RWA_CLIENTID+"&redirect_uri="+
+                process.env.REACT_APP_BACKEND_SERVER_API+
+                "&state=STATE&connection=github" || 
+                "https://torus-test.auth0.com/authorize?scope=openid&response_type=code&client_id=FX0BwYwDtD6p0yTOjjIykjLbtxXszfkR&redirect_uri=https://auth0-web3auth-example.herokuapp.com/callback&state=STATE&connection=github"
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/src/components/Main.tsx:14]: || condition is unnecessary. Also, write clean code plz.

File: examples/auth0-react-example/app/src/services/web3auth.tsx [hunk 11]
```diff
@@ -144,13 +179,34 @@ export const Web3AuthProvider: FunctionComponent<IWeb3AuthState> = ({ children,
...
+        console.log("web3auth not initialized yet");
+        uiConsole("web3auth not initialized yet");
+        return;
+      }
+      const localProvider = await web3Auth.connectTo(adapter, { 
+        relogin: true,
+        loginProvider, 
+        extraLoginOptions: {
+          id_token: jwt_token,
+          domain: "https://torus-test.auth0.com",
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/src/services/web3auth.tsx:203]: keep all auth0 and verifier conf in env variables as well. Some conf you are loading from env but  some is still static.

File: examples/auth0-react-example/app/src/services/web3auth.tsx [hunk 12]
```diff
@@ -165,6 +221,8 @@ export const Web3AuthProvider: FunctionComponent<IWeb3AuthState> = ({ children,
     }
     await web3Auth.logout();
     setProvider(null);
+    window.location.href = '/';
+    window.sessionStorage.clear();
```
Comment (himanshuchawla009) [inline examples/auth0-react-example/app/src/services/web3auth.tsx:225]: clear storage before redirecting.

**Review by himanshuchawla009 — CHANGES_REQUESTED** [review]

File: examples/auth0-react-example/app/src/services/web3auth.tsx (hunk 12 above)
Comment (rinzler99) [reply examples/auth0-react-example/app/src/services/web3auth.tsx:225]: done

File: examples/auth0-react-example/app/src/services/web3auth.tsx (hunk 11 above)
Comment (rinzler99) [reply examples/auth0-react-example/app/src/services/web3auth.tsx:203]: done

File: examples/auth0-react-example/app/src/components/Main.tsx (hunk 10 above)
Comment (rinzler99) [reply examples/auth0-react-example/app/src/components/Main.tsx:14]: done

File: examples/auth0-react-example/api/.env.example (hunk 9 above)
Comment (rinzler99) [reply examples/auth0-react-example/api/.env.example:4]: done

**Review by himanshuchawla009 — APPROVED** [review]
lgtm

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**